backend/general_functions.py

```diff
@@ -0,0 +1,372 @@
+        return False
+
+    conn = get_conn()
+    cursor = conn.cursor()
+
+    temp = ""
+    for k, v in datas.items():
+        if isinstance(v, str):
+            temp += "{} = '{}',".format(k, v)
+        else:
+            temp += "{} = {},".format(k, v)
+    temp = temp[:-1]
+
+    temp2 = ""
+    for k, v in ids.items():
+        if isinstance(v, str):
+            temp2 += "{} = '{}' and ".format(k, v)
+        else:
+            temp2 += "{} = {} and ".format(k, v)
+    temp2 = temp2[:-5]
+
+    query = ("UPDATE {} SET {} WHERE {};").format(table, temp, temp2)
+    
+    cursor.execute(query)
+
+    cursor.close()
+    conn.commit()
+    conn.close()
+
+    return True
+
+# 7- set new alert for a specific region
+def get_previous_alert(data):
+    conn = get_conn()
+    cursor = conn.cursor()
+
+    query = ("""SELECT DISTINCT alertLevel, date
+FROM alert
+WHERE regionName = '{}'
+ORDER BY date DESC
+LIMIT 1""").format(data)
+
+    cursor.execute(query)
+
+    columns = [col[0] for col in cursor.description]
+    rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+
+    cursor.close()
+    conn.commit()
+    conn.close()
+
+    temp = 0
+    for i in rows:
+        temp = i['alertLevel']
+
+    return temp
+
+def set_region_alert(data):
+    previous_alert = get_previous_alert(data['regionName'])
+
+    if data['alertLevel'] <= 0 or data['alertLevel'] > 4:
+        raise Exception('invalid alert level')
+    
+    if (data['alertLevel'] + 1 is not previous_alert) and (data['alertLevel'] - 1 is not previous_alert):
+        raise Exception('cannot set alert to this level')
+
+    create('alert', data)
+
+# 13- list of all regions
+def get_regions():
+    rows = get_all('region')
+    for region in rows:
+        cities = get_single('city', {'regionName' : region['name']})
+
+        for city in cities:
+            postal_codes = get_single('cityPostalCodeMapping', {'city' : city['name']})
+            temp = list(map(lambda x: x['postalCodeRegion'], postal_codes))
+            city['postalCodes'] = temp
+            city.pop('regionName', None)
+
+        region['cities'] = cities
+        region.pop('alertLevel', None)
+
+    return rows
+
+# 14- list of people who got the result of the test on a specific date
+def get_test_result_on_date(data):
+    conn = get_conn()
+    cursor = conn.cursor()
+    
+    query = ("""
+    SELECT P.firstName, P.lastName, P.dateOfBirth, P.phoneNumber, P.email, D.result
+FROM
+(SELECT medicareNumber, result
+FROM diagnostic
+WHERE testDate = '{}') as D, person as P
+WHERE P.medicareNumber = D.medicareNumber
+ORDER BY D.result DESC
+;""").format(data['testDate'])
+
+    cursor.execute(query)
+
+    columns = [col[0] for col in cursor.description]
+    rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+
+    cursor.close()
+    conn.commit()
+    conn.close()
+
+    return rows
+
+# 15- list of workers in specific facility
+def get_workers_at_facility(data):
```

Comment:
DB structure for workers and hospitals has changed, this will need to be updated